src/ZfrCors/Mvc/CorsRequestListener.php

```diff
@@ -112,7 +112,18 @@ public function onCorsRequest(MvcEvent $event)
         /** @var $response HttpResponse */
         $response = $event->getResponse();
 
-        if (!$request instanceof HttpRequest || !$this->corsService->isCorsRequest($request)) {
+
+        if (!$request instanceof HttpRequest) {
+            return;
+        }
+
+        // Also ensure that the vary header is set when no origin is set
+        // to prevent reverse proxy caching a wrong request; causing all of the following
+        // requests to fail due to missing CORS headers.
+        if (!$this->corsService->isCorsRequest($request)) {
+            if (!$request->getHeader("Origin")) {
```

Comment:
Can you please single quotes here (really just for consistency with the codebase).